packages/cozy-realtime/src/CozyRealtime.js

```diff
@@ -1,360 +1,534 @@
+    // if this was the last subscription, stop the realtime
+    if (this.subscriptions.isEmpty()) this.stop()
   }
 
   /**
-   * Create a Socket with cozyClient credential
+   * Unsubscribe to all events
    */
-  _createSocket() {
-    if (!this._socket) {
-      const getUrl = () => getWebSocketUrl(this._cozyClient)
-      const getToken = () => getWebSocketToken(this._cozyClient)
-
-      this._socket = new Socket(getUrl, getToken)
-      this._socket.on('message', this._receiveMessage)
-      this._socket.on('error', this._receiveError)
+  unsubscribeAll() {
+    const all = this.subscriptions.getAll()
+    for (const { event, type, id, handler } of all) {
+      this.unsubscribe(event, type, id, handler)
     }
   }
 
   /**
-   * When socket send error it test to reconnect
+   * Send all subscriptions to the current socket
+   * @private
    */
-  _receiveError(error) {
-    logger.info(`Receive error: ${error}`)
+  sendSubscriptions() {
+    const pairs = this.subscriptions.getAllTypeAndIdPairs()
+    for (const { type, id } of pairs) {
+      this.sendSubscription(type, id)
+    }
+  }
 
-    this._resetSocket()
+  /**
+   * send subscription to the websocket
+   *
+   * @private
+   * @param {string} type - a doctype to subscribe to
+   * @param {string} id - optional, a id to subscribe to
+   */
+  sendSubscription(type, id) {
+    if (this.isWebSocketOpen()) {
+      const payload = id ? { type, id } : { type }
+      logger.debug('send subscription to', payload.type, payload.id)
+      this.sendWebSocketMessage('SUBSCRIBE', payload)
+    } else {
+      logger.warn(
+        'Trying to subscribe on a non-ready socket. This should not happen. Type and id:',
+        type,
+        id
+      )
+    }
+  }
 
-    if (this._retryLimit === 0) {
-      this.emit('error', error)
+  /**
+   * send unsubscription to the websocket
+   *
+   * @private
+   * @param {string} type - a doctype to unsubscribe to
+   * @param {string} id - optional, a id to unsubscribe to
+   */
+  sendUnsubscription(type, id) {
+    if (this.isWebSocketOpen()) {
+      const payload = id ? { type, id } : { type }
+      logger.debug('send unsubscription to', payload.type, payload.id)
+      this.sendWebSocketMessage('UNSUBSCRIBE', payload)
     } else {
-      if (this.retry) {
-        clearTimeout(this.retry)
-      }
-      if (global.navigator.onLine) {
-        this.retry = setTimeout(this._resubscribe, this._retryDelay)
-      }
+      logger.warn(
+        'Trying to subscribe on a non-ready socket. This should not happen. Type and id:',
+        type,
+        id
+      )
     }
   }
 
   /**
-   * Re subscribe on server
+   * Normalize a subscription request object
+   *
+   * This allows to support the two different signatures possible
+   * for the subscribe() and the unsubscribe() methods
+   *
+   * @private
+   * @param {EventName} event
+   * @param {string} type
+   * @param {string|undefined|function} id - (or handler is `handler` is undefined)
```

Comment:
?

Reply:
Ok, I add a link to  normalizeSubscription() which  deals with that.  These methods had a special signature where you can pass (event, type, id, handler) or (event,  type,handler).I kept these possibilities for compatibility